### ISSUE TITLE
Spyder does not fit to its window

### ISSUE THREAD

**Issue by ilayn** [spyder-ide/spyder#4821]
## Description of your problem
Spyder window is rendered larger than its own containing window resulting menus and status bar to be partially displayed. However, the mouse position is still recorded correctly. 
**What steps will reproduce the problem?**

1. Straightforward installation. No actions necessary.

**What is the expected output? What do you see instead?**
This is what it looks like, notice how menu bar is folded under the containing window, also the fonts became blurry as if the whole window is scaled up (compare with the font in the menubar).

![image](https://user-images.githubusercontent.com/1303842/28646045-fbe19c0c-725f-11e7-8a94-43315ee80bc7.png)

However mouse position leads to the actual menu items notice the mark of the mouse and the hovered button location difference. 

![image](https://user-images.githubusercontent.com/1303842/28646116-5f2d9a18-7260-11e7-9858-7e4548a62d1f.png)

**Please provide any additional information below**
I am not sure if related but this started after I cloned the python-notebook plugin and installed via 
```
pip install .
```
Then I got an error on one of my notebooks and around that time on I'm having this issue.

## Versions and main components

* Spyder Version: 3.2.0
* Python Version: 3.6.1
* Qt Version: 5.9.1
* PyQt Version: 5.9
* Operating system: Windows 10

## Dependencies
```
IPython >=4.0    :  6.1.0 (OK)
cython >=0.21    :  0.26 (OK)
jedi >=0.9.0     :  0.10.2 (OK)
nbconvert >=4.0  :  5.2.1 (OK)
numpy >=1.7      :  1.13.0 (OK)
pandas >=0.13.1  :  0.20.3 (OK)
pycodestyle >=2.3:  2.3.1 (OK)
pyflakes >=0.6.0 :  1.5.0 (OK)
pygments >=2.0   :  2.2.0 (OK)
pylint >=0.25    :  1.7.2 (OK)
qtconsole >=4.2.0:  4.3.0 (OK)
rope >=0.9.4     :  0.10.5 (OK)
sphinx >=0.6.6   :  1.6.3 (OK)
sympy >=0.7.3    :  1.1 (OK)
```

**Comment by ccordoba12** [comment]
Are you using a high resolution screen?

**Comment by ilayn** [comment]
Yes I guess so. But I've tweaked the DPI settings in the preferences with no change.

**Comment by ilayn** [comment]
One thing I've noticed is that if I go into Full Screen mode, things get back to normal but the issue #4291 appears.

**Comment by ccordoba12** [comment]
I have no idea why this is happening, sorry.

**Comment by ccordoba12** [comment]
Closing because there's nothing we can do about this.

Perhaps if you use Anaconda (instead of a custom Python installation) you could have better luck.

**Comment by ilayn** [comment]
So in case if anyone stumbles upon this: 

I have no idea how it is related but today I've just turned the help pane off and it got back to normal. Before I have turned on/off quite a few of the panes but only this one forced Spyder to render properly and since then things are working properly.

**Comment by cbclass** [comment]
@ilayn  Thank you so much! This just worked for me too! The whole thing was driving me a little crazy as the mouse was displayed below so that I even had issues when positioning it in a line! Thank you! :-)

**Comment by ilayn** [comment]
@cbclass Glad to be of use. It also drove me crazy for a few months. I still don't have an answer why it happened and why this fixes the problem. Did you install the python-notebook plugin or standalone QT by any chance ?

**Comment by cbclass** [comment]
@ilayn , no I just installed the standard version on my notebook (WinPython 3.5.4.1Qt5-64bit)
. I deinstalled it and reinstalled  different versions and it did not work. More weird was the following effect: for some time, after clicking on a menu item (e.g. open file) the display was okay again. Then this stopped suddenly. But this affected mainly the frames and navigation in the beginning. But then the mouse was also wrongly displayed in the code editor which means I hardly could use it for marking code. As I use it for teaching (fortunately we did not encounter these problems in the lab) I could not just switch to another editor in order to use the same IDE in the lectures that students use in the lab.....  For two days I have seen a warning that a file is not found when I start spyder, but there is not further information regarding  the filename....

**Comment by jasonsross** [comment]
@ilayn Thank you for figuring this out! I'm on Spyder 3.2.8 and this is still an issue. I can't believe that mod closed this issue before the solution came up saying "there's nothing we can do" Hey guys at spyder, your help pane is messed up! Fix it.

**Comment by ilayn** [comment]
@jasonsross Mods are not to blame. This is not strictly a Spyder issue. Some interaction inside PyQt5 is affecting this. Cross-platforms are really tough to debug and maintaining such a project is a monster undertaking. 

That's why I'm leaving these issues just in case people like you stumble upon them and hopefully find a way that mods have no time to investigate. Check out another issue #3226 that I've left haphazardly and what it actually turns out to be. You might have the wrong impression about open-source development. It is voluntary and very very difficult (I'm on SciPy team so I know the pain). I'd prefer them coming up with new features rather than digging into this problem.

**Comment by jasonsross** [comment]
@ilayn That's fair enough, I can understand your point.

**Comment by relaxli00** [comment]
I'm using 3.2.8 within Anaconda and have the laptop screen with high resolution. I'm also using external monitors with normal resolution connected with the laptop. I suffered the exact same problem described here on either monitors. Also, the closing help-pane fixed the display problem. Thanks ilayn for your great share!

**Comment by Travis0301** [comment]
@ilayn Thank you so much! You solved my problem as well. I had an issue with the Spyder where it will only show black screen while moving mouse over the control panel will show the options. I tried to install the Spyder with different providers such as WinPython and Acaconda. as well as reconfiguring my computer settings. And I gave up after spending few days on it. But now it's all solved. Thank you again.

**Comment by elpilot** [comment]
I'm using 3.2.8 with Anaconda and also have external monitor with a laptop. Instead of closing the Help panel, in the parameters of the Help panel just select "raw text". It fixed the problem for me.

**Comment by jasonsross** [comment]
@elpilot has hit the nail on the head. The problem is using Rich test in the help pane. Turn that back to plain text and everything you can now use the help menu with everything fitting into the screen outside of fullscreen mode (you still have to go into full screen and then back out to get it right). Only problem now is Spyder is not saving that I want my help menu to be plain text...

**Comment by hubtub2** [comment]
I had the same issue here on Windows 10. Closing the help pane helped. Thank you. 
Note that not only the menu was hidden under the title bar, but the whole window looked "scaled" with terrible font and the mouse position did not match the actual "click" position.

**Comment by ccordoba12** [comment]
@jasonsross, that's because you're using Qt 5.6 instead of 5.9

**Comment by jasonsross** [comment]
@ccordoba12 ok cool. That's the first time I've seen someone say for sure Qt 5.6 will work vs 5.9 for this issue. Not sure what caused my rollback but I'll stick to it for now since it's less frustrating to deal with.

**Comment by ybilodid** [comment]
Hi!
I had exactly the same issue, and it was "cured" turning the help panel off.
But I also had a similar problem with other apps - e.g. Viber on desktop.
Now I found a real solution:
Update your video card driver!

Now both Spyder and Viber work as should,

**Comment by ccordoba12** [comment]
@ybilodid, nice!! Thanks for letting us know about it!

What video card are you using?

**Comment by ybilodid** [comment]
@ccordoba12 
It is Intel HD integrated into Core i7-7700. OS Win 10 LTSB.

**Comment by solarjoe** [comment]
This is probably the following setting in `spyder.ini`, right?

```
[help]
rich_mode = True
```

But even if I set it to `False` before starting Spyder it is ignored and changed back to `True`.
Ideas?

**Comment by ccordoba12** [comment]
@jasonsross, our next version (3.3.2) will use software rendering by default instead of accessing the video card and that could help. But we're uncertain about it.

**Comment by ilayn** [comment]
How do you mean no way to set to plain text?

**Comment by jasonsross** [comment]
> How do you mean no way to set to plain text?

@ilayn no way to permanently set it. Next time I launch spyder, the help pane is rich text again to start.

**Comment by solarjoe** [comment]
Same for me with 
```
[help]
rich_mode = True 
```
in the settings, as I mentioned above.

**Comment by drc-bsorenson** [comment]
@ccordoba12 Video driver was the issue for me, too. 
Intel HD 620 integrated into Core i5-7200. OS Win 10

**Comment by goanpeca** [comment]
We should probably not force the rich text mode on the help pane on a restart, but just use what was left before closing so that these manual tweaks are possible.

**Comment by ccordoba12** [comment]
This was fixed when we introduced the option to select the OpenGL backend used by Qt in

    Preferences > General > Rendering Engine

and set the default value of that option to `Software`.
